.github/workflows/python-app.yml

```diff
@@ -27,11 +27,11 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-      - name: Lint with pylint
+          python -m pip install uv
+          if [ -f requirements.txt ]; then uv pip install -r requirements.txt --system; fi
+      - name: Lint with ruff
         run: |
-          pylint packet/routes packet
+          ruff check packet
```

Comment:
Looking at the [ruff FAQ](https://docs.astral.sh/ruff/faq/#how-does-ruffs-linter-compare-to-pylint), seems like they're more confident in how well it can replace pylint than a few years ago, but it also discuss that they can be complimentary. Would it make sense to use ruff locally for anyone using it for commit hooks, and pylint in CI for completeness